### DISCUSSION TITLE
How to customize font icons (with text) in NavigationStore Items?

### DISCUSSION THREAD

**Discussion by ghost** [lepoco/wpfui#448] [Q&A]
I want to fully customize the icons provided by default in the NavigationStore Items. I succeeded in rendering a glyph, by using custom font. But I need to show both text and matching filled icon for each item in the main navigation of the application, either. How can I achieve that?

Image of non selected icon (missing text):
<img width="68" alt="image" src="https://user-images.githubusercontent.com/46963026/198721265-d5245de9-e645-440a-8e7c-f3ac1a7a35c0.png">

Image of active icon (missing both text and glyph):
<img width="84" alt="image" src="https://user-images.githubusercontent.com/46963026/198721430-77dec050-d29f-436b-bd39-e4de250b298a.png">

Code used so far:
```
<ui:NavigationStore
                x:Name="RootNavigation"
                Frame="{Binding ElementName=RootFrame, Mode=OneWay}">
                <ui:NavigationStore.Items>
                    <ui:NavigationItem
                          Content="Home"
                          Icon="Home24"
                          PageType="{x:Type pages:DashboardPage}"
                          PageTag="dashboard" />
                    <ui:NavigationItem
                        Height="64"
                        PageTag="data"
                        IconSize="24"
                        PageType="{x:Type pages:DataPage}"
                        VerticalAlignment="Center" HorizontalAlignment="Center">
                            <ui:FontIcon
                                Content="data"
                                Margin="6,-35,0,0"
                                VerticalAlignment="Center" HorizontalAlignment="Center"
                                Width="32"
                                Height="32"
                                MinWidth="32"
                                MinHeight="32"
                                FontFamily="{DynamicResource TryFontIcons}"
                                FontSize="24"
                                Glyph="&#xe900;"/>
                    </ui:NavigationItem>
                </ui:NavigationStore.Items>
            </ui:NavigationStore>
```

**Comment by pomianowski** [comment]
NavigationView does not support custom icons and this feature is not planned yet. May (or not) be added in the future.
The solution to suit your idea is to create your own style or custom control which would extend NavigationView.

Additionally.
The reason the active item doesn't have an icon is because the regular item uses `SymbolRegular` and the active one uses `SymbolFilled`. Which are two separate fonts.
